### ISSUE TITLE
recipe for target 'Deepin/gala-DeepinWindowSwitcherItem.o' failed

### ISSUE THREAD

**Issue by yunusem** [linuxdeepin/deepin-wm#25]
```
Deepin/DeepinWindowSwitcherItem.c:175:43: error: stray ‘\304’ in program
  GALA_DEEPIN_WINDOW_SWITCHER_ITEM_PREFER_WıDTH,
                                           ^
Deepin/DeepinWindowSwitcherItem.c:175:44: error: stray ‘\261’ in program
  GALA_DEEPIN_WINDOW_SWITCHER_ITEM_PREFER_WıDTH,
                                            ^
Deepin/DeepinWindowSwitcherItem.c:175:45: error: expected ‘,’ or ‘}’ before ‘DTH’
  GALA_DEEPIN_WINDOW_SWITCHER_ITEM_PREFER_WıDTH,
                                             ^~~
Deepin/DeepinWindowSwitcherItem.c:176:44: error: stray ‘\304’ in program
  GALA_DEEPIN_WINDOW_SWITCHER_ITEM_PREFER_HEıGHT,
                                            ^
Deepin/DeepinWindowSwitcherItem.c:176:45: error: stray ‘\261’ in program
  GALA_DEEPIN_WINDOW_SWITCHER_ITEM_PREFER_HEıGHT,
                                             ^
Deepin/DeepinWindowSwitcherItem.c:177:48: error: stray ‘\304’ in program
  GALA_DEEPIN_WINDOW_SWITCHER_ITEM_RECT_PREFER_WıDTH,
                                                ^
Deepin/DeepinWindowSwitcherItem.c:177:49: error: stray ‘\261’ in program
  GALA_DEEPIN_WINDOW_SWITCHER_ITEM_RECT_PREFER_WıDTH,
                                                 ^
Deepin/DeepinWindowSwitcherItem.c:178:49: error: stray ‘\304’ in program
  GALA_DEEPIN_WINDOW_SWITCHER_ITEM_RECT_PREFER_HEıGHT,
                                                 ^
Deepin/DeepinWindowSwitcherItem.c:178:50: error: stray ‘\261’ in program
  GALA_DEEPIN_WINDOW_SWITCHER_ITEM_RECT_PREFER_HEıGHT,
                                                  ^
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala: In function ‘gala_deepin_window_switcher_item_class_init’:
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:100: error: stray ‘\304’ in program
  public class DeepinWindowSwitcherItem : Actor
                                                                                                    ^
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:101: error: stray ‘\261’ in program
  public class DeepinWindowSwitcherItem : Actor
                                                                                                     ^
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:102: error: expected ‘)’ before ‘DTH’
  public class DeepinWindowSwitcherItem : Actor
                                                                                                      ^  
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:2: error: too few arguments to function ‘g_object_class_install_property’
  public class DeepinWindowSwitcherItem : Actor
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/gobject/gbinding.h:29:0,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from Deepin/DeepinWindowSwitcherItem.c:22:
/usr/include/glib-2.0/gobject/gobject.h:389:13: note: declared here
 void        g_object_class_install_property   (GObjectClass   *oclass,
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:101: error: stray ‘\304’ in program
  public class DeepinWindowSwitcherItem : Actor
                                                                                                     ^
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:102: error: stray ‘\261’ in program
  public class DeepinWindowSwitcherItem : Actor
                                                                                                      ^
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:59: error: ‘GALA_DEEPIN_WINDOW_SWITCHER_ITEM_PREFER_HE’ undeclared (first use in this function)
  public class DeepinWindowSwitcherItem : Actor
                                                           ^                                         
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:59: note: each undeclared identifier is reported only once for each function it appears in
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:103: error: expected ‘)’ before ‘GHT’
  public class DeepinWindowSwitcherItem : Actor
                                                                                                       ^  
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:2: error: too few arguments to function ‘g_object_class_install_property’
  public class DeepinWindowSwitcherItem : Actor
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/gobject/gbinding.h:29:0,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from Deepin/DeepinWindowSwitcherItem.c:22:
/usr/include/glib-2.0/gobject/gobject.h:389:13: note: declared here
 void        g_object_class_install_property   (GObjectClass   *oclass,
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:105: error: stray ‘\304’ in program
  public class DeepinWindowSwitcherItem : Actor
                                                                                                         ^
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:106: error: stray ‘\261’ in program
  public class DeepinWindowSwitcherItem : Actor
                                                                                                          ^
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:59: error: ‘GALA_DEEPIN_WINDOW_SWITCHER_ITEM_RECT_PREFER_W’ undeclared (first use in this function)
  public class DeepinWindowSwitcherItem : Actor
                                                           ^                                             
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:107: error: expected ‘)’ before ‘DTH’
  public class DeepinWindowSwitcherItem : Actor
                                                                                                           ^  
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:2: error: too few arguments to function ‘g_object_class_install_property’
  public class DeepinWindowSwitcherItem : Actor
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/gobject/gbinding.h:29:0,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from Deepin/DeepinWindowSwitcherItem.c:22:
/usr/include/glib-2.0/gobject/gobject.h:389:13: note: declared here
 void        g_object_class_install_property   (GObjectClass   *oclass,
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:106: error: stray ‘\304’ in program
  public class DeepinWindowSwitcherItem : Actor
                                                                                                          ^
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:107: error: stray ‘\261’ in program
  public class DeepinWindowSwitcherItem : Actor
                                                                                                           ^
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:59: error: ‘GALA_DEEPIN_WINDOW_SWITCHER_ITEM_RECT_PREFER_HE’ undeclared (first use in this function)
  public class DeepinWindowSwitcherItem : Actor
                                                           ^                                              
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:108: error: expected ‘)’ before ‘GHT’
  public class DeepinWindowSwitcherItem : Actor
                                                                                                            ^  
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:2: error: too few arguments to function ‘g_object_class_install_property’
  public class DeepinWindowSwitcherItem : Actor
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/gobject/gbinding.h:29:0,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from Deepin/DeepinWindowSwitcherItem.c:22:
/usr/include/glib-2.0/gobject/gobject.h:389:13: note: declared here
 void        g_object_class_install_property   (GObjectClass   *oclass,
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:59: error: ‘GALA_DEEPIN_WINDOW_SWITCHER_ITEM_SCALE_FACTOR’ undeclared (first use in this function)
  public class DeepinWindowSwitcherItem : Actor
                                                           ^                                            
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala: In function ‘_vala_gala_deepin_window_switcher_item_get_property’:
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:49: error: stray ‘\304’ in program
  public class DeepinWindowSwitcherItem : Actor
                                                 ^
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:50: error: stray ‘\261’ in program
  public class DeepinWindowSwitcherItem : Actor
                                                  ^
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:51: error: expected ‘:’ or ‘...’ before ‘DTH’
  public class DeepinWindowSwitcherItem : Actor
                                                   ^  
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:50: error: stray ‘\304’ in program
  public class DeepinWindowSwitcherItem : Actor
                                                  ^
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:51: error: stray ‘\261’ in program
  public class DeepinWindowSwitcherItem : Actor
                                                   ^
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:8: error: ‘GALA_DEEPIN_WINDOW_SWITCHER_ITEM_PREFER_HE’ undeclared (first use in this function)
  public class DeepinWindowSwitcherItem : Actor
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~   
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:52: error: expected ‘:’ or ‘...’ before ‘GHT’
  public class DeepinWindowSwitcherItem : Actor
                                                    ^  
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:54: error: stray ‘\304’ in program
  public class DeepinWindowSwitcherItem : Actor
                                                      ^
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:55: error: stray ‘\261’ in program
  public class DeepinWindowSwitcherItem : Actor
                                                       ^
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:8: error: ‘GALA_DEEPIN_WINDOW_SWITCHER_ITEM_RECT_PREFER_W’ undeclared (first use in this function)
  public class DeepinWindowSwitcherItem : Actor
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~       
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:56: error: expected ‘:’ or ‘...’ before ‘DTH’
  public class DeepinWindowSwitcherItem : Actor
                                                        ^  
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:56: error: duplicate label ‘DTH’
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:51: note: previous definition of ‘DTH’ was here
  public class DeepinWindowSwitcherItem : Actor
                                                   ^  
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:55: error: stray ‘\304’ in program
  public class DeepinWindowSwitcherItem : Actor
                                                       ^
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:56: error: stray ‘\261’ in program
  public class DeepinWindowSwitcherItem : Actor
                                                        ^
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:8: error: ‘GALA_DEEPIN_WINDOW_SWITCHER_ITEM_RECT_PREFER_HE’ undeclared (first use in this function)
  public class DeepinWindowSwitcherItem : Actor
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~        
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:57: error: expected ‘:’ or ‘...’ before ‘GHT’
  public class DeepinWindowSwitcherItem : Actor
                                                         ^  
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:57: error: duplicate label ‘GHT’
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:52: note: previous definition of ‘GHT’ was here
  public class DeepinWindowSwitcherItem : Actor
                                                    ^  
/tmp/build-area/deepin-wm-1.9.21/src/Deepin/DeepinWindowSwitcherItem.vala:26:8: error: ‘GALA_DEEPIN_WINDOW_SWITCHER_ITEM_SCALE_FACTOR’ undeclared (first use in this function)
  public class DeepinWindowSwitcherItem : Actor
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~      
Makefile:1150: recipe for target 'Deepin/gala-DeepinWindowSwitcherItem.o' failed
make[4]: *** [Deepin/gala-DeepinWindowSwitcherItem.o] Error 1
make[4]: *** Bitmemiş işler için bekliyor....
make[4]: Leaving directory '/tmp/build-area/deepin-wm-1.9.21/src'
Makefile:516: recipe for target 'all' failed
make[3]: *** [all] Error 2
make[3]: Leaving directory '/tmp/build-area/deepin-wm-1.9.21/src'
Makefile:491: recipe for target 'all-recursive' failed
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory '/tmp/build-area/deepin-wm-1.9.21'
Makefile:422: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/tmp/build-area/deepin-wm-1.9.21'
dh_auto_build: make -j3 returned exit code 2
debian/rules:4: recipe for target 'build' failed
make: *** [build] Error 2
dpkg-buildpackage: error: debian/rules build gave error exit status 2
debuild: fatal error at line 1116:
dpkg-buildpackage -rfakeroot -us -uc -i -I -b failed
gbp:error: 'debuild -i -I -b -us -uc' failed: it exited with 29
```

I'm building from source and trying to implement the package in [Pardus](https://github.com/Pardus-Kurumsal?utf8=%E2%9C%93&q=dde&type=&language=) Distro 

Regards

**Comment by yunusem** [comment]
Any help about this ?

**Comment by sonald** [comment]
@yunusem Can you upload full compilation process? what vala version did you use and what is your locale ? TR.UTF-8?

**Comment by yunusem** [comment]
$ dpkg -l valac
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                                                           Version                              Architecture                         Description

ii  valac                                                          0.34.7-1                             amd64                                C# like language for the GObject system

I am working on vagrant pardus.
Problem **solved** with **valac 0.34.7-1**, i think build dependency should be updated with:
valac (>= 0.26.1) against valac (>= 0.22.0)

Thanx for quick reply.

**Comment by sonald** [comment]
that makes sense. we do need higher version of vala, gonna update it.
